spark/src/main/java/org/apache/iceberg/spark/actions/BaseRewriteDataFilesSparkAction.java

```diff
@@ -383,4 +408,31 @@ public int totalGroupCount() {
       return totalGroupCount;
     }
   }
+
+  static class EmptyStruct implements StructLike {
+
+    private static EmptyStruct instance;
+
+    static EmptyStruct instance() {
+      if (instance == null) {
+        instance = new EmptyStruct();
+      }
+      return instance;
+    }
+
+    @Override
+    public int size() {
+      return 0;
+    }
+
+    @Override
+    public <T> T get(int pos, Class<T> javaClass) {
+      return null;
+    }
+
+    @Override
+    public <T> void set(int pos, T value) {
+
```

Comment:
Should this throw an exception because you're trying to write to a slot that doesn't exist?